saltstack/base/salt/ccm-client/cdp/bin/reverse-tunnel.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash -ux
+
+SCRIPT=`basename "$0"`
+ROLE=$1
+
+. /cdp/bin/reverse-tunnel-values-${ROLE}.sh
+
+# Setting gate time to 0 to retry on SSH failures due to connectivity issues on first try
+export AUTOSSH_GATETIME=0
+export AUTOSSH_LOGLEVEL=7
+export AUTOSSH_LOGFILE="/var/log/autossh-${ROLE}.log"
+# Set max backoff interval in case of recurring connection failures.
+export AUTOSSH_POLL=30
+
+PRIVATE_KEY=/tmp/pk.key
+
+if grep -q "BEGIN" ${ENCIPHERED_PRIVATE_KEY}; then
+    cat ${ENCIPHERED_PRIVATE_KEY} > ${PRIVATE_KEY}
+else
```

Comment:
Do we have such a use-case when there is unencrypted ENCIPHERED_PRIVATE_KEY?

Reply:
Well technically in all production usages, we should be using Enciphered value, But in testing and debugging, depending on the team doing the work, they might want to use the plain key as is. That was the usecase. Also, Altus originally supported unencrypted and for backward incompatibility, we kept it as is.

Reply:
Peter, I will resolve this conversation. Reopen if you still have concerns.